### PR TITLE
Fix #2845 - Init Profiler Workflow

### LINKED ISSUE #2845: Profiler - Define workflow
**Feature**
ORM Profiler for data quality https://github.com/open-metadata/OpenMetadata/issues/1311

**Describe the task**
Now that we've defined the core entities, we need to prepare the end-to-end process. Let's define a separated workflow where we can:
- define the source
- filter entities to be run
- select metrics (or run simple profiler by default)
- define tests

Also, let's try to:
- test custom registries
- add missing types
- Allow spaces in validation values grammar

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR fixes https://github.com/open-metadata/OpenMetadata/issues/2845

We have now prepared a separated profiler workflow based on the structure and sources from the ingestion one.

The approach is:
- List table entities from OM
- Filter tables based on the service
- Apply table and schema filter patterns as in the ingestion step
- Run the table & column profilers. If no profiler is specified, we default to our own profilers.
- Run the validations & report back the results.

We modified a bit the core definitions to ease the workflow creation. Based on the config dict, we can directly validate the test information and parse the expressions, so if there are any issues we do not need to wait for the workflow to run. This will also allow users to even run tests on their own and have fast feedback.

Further work is pending to have a similar feedback structure as in the ingestion process and check all edge cases (e.g., a test is defined with a metric not selected for the profiler, this should skip the validation and let the process continue).

Also, we still need to define the profiler sink to push the results back to OM.

> OBS: For testing purposes, we added the `SQLite` connector. We can then run the profiler workflow integration tests just by having the OM server running and without having to configure any other database. I'll iterate more on these tests to be more thorough.

Thanks!

#
### Type of change :

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/orm_profiler/validations/models.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        Make sure that we can parse the expression to a
+        validation
+        """
+        try:
+            raw_validation = parse(value, visitor)
+        except ParseError as err:
+            logger.error(f"Cannot parse expression properly: {value}")
+            raise err
+
+        return [Validation.create(val) for val in raw_validation]
```
Comment (pmbrull) [inline ingestion/src/metadata/orm_profiler/validations/models.py:53]: This allows us to create directly the Validation object when parsing the profiler config

File: ingestion/src/metadata/config/workflow.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        my_class = getattr(importlib.import_module(module_name), class_name)
+        return my_class
+
+
+def get_ingestion_source(
+    source_type: str,
+    context: WorkflowContext,
+    source_config: DynamicTypedConfig,
+    metadata_config: MetadataServerConfig,
+) -> Source:
```
Comment (pmbrull) [inline ingestion/src/metadata/config/workflow.py:57]: These methods have been extracted from the ingestion workflow definition. When tackling https://github.com/open-metadata/OpenMetadata/issues/2848, we will then simplify a bit the Ingestion Workflow to use them as well.

File: ingestion/src/metadata/ingestion/ometa/ometa_api.py [hunk 3]
```diff
@@ -449,7 +449,7 @@ def list_entities(
         url_limit = f"?limit={limit}"
         url_after = f"&after={after}" if after else ""
         url_fields = f"&fields={','.join(fields)}" if fields else ""
-        url_params = f"&{urllib.parse.urlencode(params)}"
+        url_params = f"&{urllib.parse.urlencode(params)}" if params else ""
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/ometa_api.py:452]: Small sanity fix

File: ingestion/src/metadata/orm_profiler/validations/models.py (hunk 1 above)
Comment (harshach) [reply ingestion/src/metadata/orm_profiler/validations/models.py:53]: @pmbrull once the APIs are ready we should integrate the test case validation parsing from the table apis

**Review by harshach — APPROVED** [review]

**Comment by pmbrull** [comment]
Flaky webhook test. Merging! Thanks

**Pull request merged**